socorro/app/fetch_transform_save_app.py

```diff
@@ -37,180 +37,71 @@
 
 
 class FetchTransformSaveApp(App):
-    """base class for apps that follow the fetch/transform/save model"""
-    app_name = 'generic_fetch_transform_save_app'
+    """Fetch/transform/save apps."""
+
+    app_name = 'fetch_transform_save_app'
     app_version = '0.1'
     app_description = __doc__
 
     required_config = Namespace()
-    # the required config is broken into two parts: the source and the
-    # destination.  Each of those gets assigned a crasnstorage class.
+    # The queue class has an iterator for work items to be processed.
+    required_config.namespace('queue')
+    required_config.queue.add_option(
+        'crashqueue_class',
+        doc='an iterable that will stream work items for processing',
+        default='',
+        from_string_converter=class_converter
+    )
+
+    # The source class has methods to fetch the data to use.
     required_config.source = Namespace()
-    # For source, the storage class should be one that defines a method
-    # of fetching new crashes through the three storage api methods: the
-    # iterator 'new_crashes' and the accessors 'get_raw_crash' and
-    # 'get_raw_dumps'
     required_config.source.add_option(
         'crashstorage_class',
         doc='the source storage class',
         default='socorro.external.fs.crashstorage.FSPermanentStorage',
         from_string_converter=class_converter
     )
+
+    # The destination class has methods to save the transformed data to storage.
     required_config.destination = Namespace()
-    # For destination, the storage class should implement the 'save_raw_crash'
-    # method.  Of course, a subclass may redefine either the source_iterator
-    # or transform methods and therefore completely redefine what api calls
-    # are relevant.
     required_config.destination.add_option(
         'crashstorage_class',
         doc='the destination storage class',
         default='socorro.external.fs.crashstorage.FSPermanentStorage',
         from_string_converter=class_converter
     )
+
     required_config.producer_consumer = Namespace()
     required_config.producer_consumer.add_option(
         'producer_consumer_class',
         doc='the class implements a threaded producer consumer queue',
         default='socorro.lib.threaded_task_manager.ThreadedTaskManager',
         from_string_converter=class_converter
     )
-    required_config.add_option(
-        'number_of_submissions',
-        doc="the number of crashes to submit (all, forever, 1...)",
-        short_form='n',
-        default='forever'
-    )
 
     def __init__(self, config):
         super().__init__(config)
         self.waiting_func = None
-        # select the iterator type based on the "number_of_submissions" config
-        self.source_iterator = {
-            'forever': self._infinite_iterator,
-            'all': self._all_iterator,
-        }.get(config.number_of_submissions, self._limited_iterator)
```

Comment:
We only use `forever` except for a few tests which I reworked. Given that, I nixed the other iterators.